backends/bmv2/common/parser.cpp

```diff
@@ -287,16 +287,23 @@ void ParserConverter::convertSimpleKey(const IR::Expression* keySet,
         auto range = keySet->to<IR::Range>();
         if (!range->left->is<IR::Constant>()) {
             ::error(ErrorType::ERR_INVALID, "must evaluate to a compile-time "
-                  "constant", range->left);
+                    "constant", range->left);
             return;
         }
+        auto left = range->left->to<IR::Constant>()->value;
         if (!range->right->is<IR::Constant>()) {
-            ::error(ErrorType::ERR_INVALID, "must evaluate to a compile-time "
-            "constant", range->right);
+            ::error(ErrorType::ERR_INVALID, "must evaluate to a compile-time"
+                    "constant", range->right);
+            return;
+        }
+        auto right = range->right->to<IR::Constant>()->value;
+        if (right <= left) {
```

Comment:
I think this should be `right < left`.  While it might be odd for someone to make the left and right equal to each other, it doesn't seem like it ought to be an error.